### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by veronica365**
### Description
- Created a class called Solver.
- Created methods called factorial, reverse and fizzbuzz
- Used TDD Approach. Commit history proves that we have used TDD.
- Following Gitflow
- Add Linters

### CODE REVIEW

File: spec/factorial_spec.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  def factorial(number)
+    factor = Solver.new
+    factor.factorial(number)
+  end
+  it 'should be able to calculate the factorial of 5' do
+    expect(factorial(5)).to eq(120)
+  end
+  it 'should be able to calculate the factorial of 0' do
+    expect(factorial(0)).to eq(1)
+  end
```
Comment (omar-muhamad) [inline spec/factorial_spec.rb:13]: - [ ] Nice job here testing the factorial method, As per this [requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md#:~:text=positive%20integers%2C%20so-,if%20a%20negative%20integer%20is%20given%20it%20should%20raise%20an%20exception.,-Create%20a%20method), Please add a test to raise an exception when you try to get the factorial of a negative number to cover all edge cases.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @veronica365 and @QueenterJuma,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
.

**Review by AnselemOdims — APPROVED** [review]
### Hi  @QueenterJuma 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
